### PR TITLE
Changed the function type of commitMetablock of Consensus contract from public to external

### LINKED ISSUE #245: Update Consensus::commitMetablock
Current implementation of `Consensus::commitMetablock` is `public`, convert it to `external`
  - The function can be broken down so as to achieve this

### PR DESCRIPTION
**Pull request by bruce9steve**
fixes #245 
- Changed the function type of commitMetablock in Consensus contract from "public" to "external".

### CODE REVIEW

File: contracts/consensus/Consensus.sol [hunk 1]
```diff
@@ -701,15 +698,62 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
...
+        uint256 _gasTargetDelta,
+        bytes32 _kernelHash
+    )
+        private
+    {
         // Anchor state root.
         anchorStateRoot(_metachainId, _rlpBlockHeader);
 
+        CoreI core = assignments[_metachainId];
+        bytes32 metablockHash = goToCommittedRound(_metachainId);
```
Comment (0xsarvesh) [inline contracts/consensus/Consensus.sol:749]: This is call twice in below functions : 
1. assertCommit
2. commitStateRoot

File: contracts/consensus/Consensus.sol [hunk 2]
```diff
@@ -701,15 +698,62 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
...
+        uint256 _sourceBlockHeight,
+        uint256 _gasTargetDelta,
+        bytes32 _kernelHash
+    )
+        private
+    {
         // Anchor state root.
         anchorStateRoot(_metachainId, _rlpBlockHeader);
 
+        CoreI core = assignments[_metachainId];
```
Comment (0xsarvesh) [inline contracts/consensus/Consensus.sol:748]: Core address is read three times. Is it possible to optimise this?

File: contracts/consensus/Consensus.sol [hunk 3]
```diff
@@ -701,15 +698,62 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
...
+     * @param _rlpBlockHeader RLP encoded block header.
+     * @param _dynasty The dynasty number where the meta-block closes
+                       on the auxiliary chain.
+     * @param _accumulatedGas The total consumed gas on auxiliary within this
+                              meta-block.
+     * @param _sourceBlockHeight Source block height.
+     * @param _gasTargetDelta Gas target delta to open new metablock.
+     * @param _kernelHash Kernel hash.
+     */
+    function commitStateRoot(
```
Comment (0xsarvesh) [inline contracts/consensus/Consensus.sol:734]: It's more readable if event is emitted by external function instead of private.

Comment (0xsarvesh) [inline contracts/consensus/Consensus.sol:734]: Naming can be improved.

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]
Thanks 👍  
Some feedback.

**Comment by bruce9steve** [comment]
As discussed with @deepesh-kn I am moving this ticket to parking lot.

**Comment by benjaminbollen** [comment]
Thanks, but I will close this.

**Pull request closed without merging**